### PR TITLE
ROE-1425 List validation errors and remove stack trace

### PR DESCRIPTION
**Pull request by mattch1**
https://companieshouse.atlassian.net/browse/ROE-1425

Used the gson library present in the project to get a readable list of errors in json format. Removed stack trace by setting the exception argument to null.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesController.java [hunk 1]
```diff
@@ -66,7 +67,8 @@ public ResponseEntity<Object> createNewSubmission(
                 var validationErrors = overseasEntitySubmissionDtoValidator.validate(overseasEntitySubmissionDto, new Errors(), requestId);
 
                 if (validationErrors.hasErrors()) {
-                    ApiLogger.errorContext(requestId, "Validation errors : " + validationErrors, new Exception());
+
+                    ApiLogger.errorContext(requestId, "Validation errors : " + convertErrorsToString(validationErrors), null);
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesController.java:71]: can you add a unit test to check the errors are passed to the logger? have a look at https://www.baeldung.com/mockito-mock-static-methods#mocking-a-static-method-with-arguments for mocking a static call and maybe use an argument captor to capture the message

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesController.java [hunk 2]
```diff
@@ -88,6 +90,11 @@ public ResponseEntity<Object> createNewSubmission(
         }
     }
 
+    private String convertErrorsToString(Errors validationErrors) {
+        var gson = new GsonBuilder().create();
+        return gson.toJson(validationErrors);
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesController.java:95]: I'm interested in what this looks like in the output. Could you please copy and paste an extract from the Docker Tilt log here?

Also, minor, but suggest renaming method to `convertErrorsToJsonString` - conveys the type of string you'll get back.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesController.java (hunk 1 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesController.java:71]: I've not heard of unit tests for error logging before but have managed to add something. However, the test is misreporting the result:

Wanted but not invoked:
ApiLogger.class.errorContext(
    <any>,
    "Validation errors : {"errs":[{"error":"Any","location":"Any","locationType":"request-body","type":"ch:validation"}]}",
    null
);
 When I debug I can see it being invoked so I am not sure what problem it is picking up on. I have committed what I have done and will require someone to check this branch out and take a look at it.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesController.java (hunk 2 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesController.java:95]: An example can be seen above.

Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesController.java:95]: From the logs:
{"created":"2022-10-25T09:23:16.310Z","event":"error","namespace":"overseas-entities-api","context":"NviWdOblk31O714ygPYjs8APsar8","data":{"message":"Validation errors : {\"errs\":[{\"error\":\"Argentina\\\\u0414\\\\u0440\\\\u0430\\\\u043a\\\\u043e\\\\u043d is not on the list of allowed countries\",\"location\":\"entity.incorporation_country\",\"locationType\":\"request-body\",\"type\":\"ch:validation\"}]}"}}

Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesController.java:95]: 👍 nice. Thanks.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesController.java (hunk 1 above)
Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesController.java]: I'll grab this and take a look now @mattch1 .

Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesController.java]: Change pushed. You just needed to move the actual test execution code into that try block, above the new static verify() check. I also made a couple of other very minor improvements. Please take a look @mattch1 .

Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesController.java]: Just run it locally, looks good, so the position of the try block was the issue?

Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesController.java]: Yes.

**Review by markpit — APPROVED** [review]

**Pull request merged**